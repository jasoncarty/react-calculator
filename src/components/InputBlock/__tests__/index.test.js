import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import { InputBlock } from "../index";

describe("<InputBlock />", () => {
  it("matches snapshot", () => {
    const instance = ReactTestRenderer.create(
      <InputBlock
        index={0}
        changeValue={() => {}}
      />
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("updates state", () => {
    const instance = ReactTestRenderer.create(
      <InputBlock
        index={1}
        changeValue={() => {}}
      />
    );

    const input = instance.root.findByProps({
      type: "text"
    });

    input.props.onChange({
      target: {
        value: "890"
      }
    });

    expect(instance.getInstance().state)
      .toEqual({
        value: "890"
      });

    input.props.onChange({
      target: {
        value: "80"
      }
    });

    expect(instance.getInstance().state)
      .toEqual({
        value: "80"
      });

    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("Calls changeValue", () => {
    const mockFn = jest.fn();
    const instance = ReactTestRenderer.create(
      <InputBlock
        index={1}
        changeValue={mockFn}
      />
    );

    const input = instance.root.findByProps({
      type: "text"
    });

    input.props.onChange({
      target: {
        value: "890"
      }
    });

    expect(mockFn).toBeCalledWith({
      index: 1,
      payload: "890"
    });

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
