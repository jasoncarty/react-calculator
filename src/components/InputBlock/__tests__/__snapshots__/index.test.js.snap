// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputBlock /> Calls changeValue 1`] = `
<div
  className="inputBlock"
>
  <div
    className="content"
  >
    <div
      className="text"
    >
      Value 2: 
    </div>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="890"
    />
  </div>
</div>
`;

exports[`<InputBlock /> matches snapshot 1`] = `
<div
  className="inputBlock"
>
  <div
    className="content"
  >
    <div
      className="text"
    >
      Value 1: 
    </div>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<InputBlock /> updates state 1`] = `
<div
  className="inputBlock"
>
  <div
    className="content"
  >
    <div
      className="text"
    >
      Value 2: 
    </div>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="80"
    />
  </div>
</div>
`;
